### PR TITLE
Fix DG features sequence diagrams

### LINKED ISSUE #224: Edit sequence diagram and activity diagram images
Change the params if they are too long + update the images to most updated implementation

### PR DESCRIPTION
**Pull request by jiahui0309**
Modified the sequence diagrams, added logging statements. Closes #224.

This PR does not include (as of latest commit):
- any changes to activity diagrams

This PR includes:
- logging
- DG sequence diagrams
- changes to DG implementation content for add, addnote, find

Screenshot 2024-04-14 at 9 50 17 PM
Screenshot 2024-04-14 at 9 48 31 PM
Screenshot 2024-04-14 at 9 48 44 PM
Screenshot 2024-04-14 at 9 49 03 PM
Screenshot 2024-04-14 at 9 49 22 PM

### CODE REVIEW

**Comment by drustanyjt** [comment]
@jiahui0309 why do the `XYZCommandParser` have crosses for object deletion but some of the other things (like the `XYZCommand` objects) do not? Should we add object deletions for the `XYZCommand` too after `r` is returned?

**Comment by jiahui0309** [comment]
@drustanyjt okay! initially i was following the diagram they provided, but it seems reasonable to delete the obj after returning the result

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -160,16 +160,24 @@ This section describes some noteworthy details on how certain features are imple
...
+Apart from that, the feature also includes the following operation in `ModelManager`, which implements the `Model` interface:
+- `Model#hasPerson(Person person)`: Checks if the `Person` to be added is already an existing `Person` profile in ClinicMate
 
-Step 1. The user launches the application for the first time. The `AddressBook` will be initialized with the initial address book state.
+Given below is an example usage scenario and how the `add` mechanism behaves at each step.
 
-Step 2. The user executes `add n\John Doe …` to add the person in the address book with the unique identification number `T0123456A`. The add command calls `Model#addPerson(Person p)`, causing the modified state of the address book after the `add n\John Doe …` command executes to be saved.
+Step 1. The user launches the application for the first time. `ClinicMate` will be initialized with the initial address book state.
 
+Step 2. The user executes `add n\John Doe p\12345678 e\john@mail.com i\T0123456A ag\12 s\M a\John street, block 123, #01-01` to add the person in the address book with the unique identification number `T0123456A`.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: maybe change identification number to IC number? to standardise

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -160,16 +160,24 @@ This section describes some noteworthy details on how certain features are imple
...
+- `Model#hasPerson(Person person)`: Checks if the `Person` to be added is already an existing `Person` profile in ClinicMate
 
-Step 1. The user launches the application for the first time. The `AddressBook` will be initialized with the initial address book state.
+Given below is an example usage scenario and how the `add` mechanism behaves at each step.
 
-Step 2. The user executes `add n\John Doe …` to add the person in the address book with the unique identification number `T0123456A`. The add command calls `Model#addPerson(Person p)`, causing the modified state of the address book after the `add n\John Doe …` command executes to be saved.
+Step 1. The user launches the application for the first time. `ClinicMate` will be initialized with the initial address book state.
 
+Step 2. The user executes `add n\John Doe p\12345678 e\john@mail.com i\T0123456A ag\12 s\M a\John street, block 123, #01-01` to add the person in the address book with the unique identification number `T0123456A`. 
+The add command calls `Model#addPerson(Person p)`, causing the modified state of the address book after the `add n/John Doe …` command executes to be saved.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: code block for add

Comment (tengcharmaine) [reply docs/DeveloperGuide.md]: add n/John... -> add n\John

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -209,48 +217,54 @@ The following activity diagram summarizes what happens when a user executes a ne
 * **Alternative 2:** Allow users to add patients by adding fields as and when is needed (i.e. not make all the fields compulsory).
     * Pros: Shorter command to type to add a patient.
     * Cons: 
-      * We must ensure that the implementation of each individual command are correct. 
-      * Might not have all the relevant information of all patients. Messy to keep track.
+      * We must ensure that the implementation of each individual commands is correct. 
+      * Might not have all the relevant information of the patients. Messy to keep track.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: i think can remove 'the'

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -209,48 +217,54 @@ The following activity diagram summarizes what happens when a user executes a ne
...
 #### Implementation
 
-The add/replace notes mechanism is facilitated by `AddressBook`. It implements `AddressBook#setPerson(Person target, Person editedPerson)` which allow users to add/replace patients’ notes in the addressbook.
+The add/replace note mechanism is facilitated by `AddressBook`. It implements `AddressBook#setPerson(Person target, Person editedPerson)` which allow users to add/replace patients’ notes in the addressbook.
 
 These operations are exposed in the `Model` interface as `Model#setPerson(Person target, Person editedPerson)`
 
+The `addnote` feature has the following operations in `ModelManager` which implements the `Model` interface:
+- `Model#setPerson`: Changes the note parameter of the target Person
+- `Model#isPersonDisplayed`: Checks if the `Person` has their notes displayed in the person notes panel
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: patient notes panel instead of person notes panel

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -209,48 +217,54 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+- `Model#setDisplayedNote`: If `Model#isPersonDisplayed` returns true, the notes displayed will be updated
+
 Given below is an example usage scenario and how the add/replace note mechanism behaves at each step.
 
 Step 1. The user launches the application. The `AddressBook` will be initialized with the initial address book state.
 
-Step 2. The user executes `addnote T0123456A …` to add a note to the person in the address book with the unique identification number `T0123456A`. The addnote command calls `Model#setPerson(Person target, Person editedPerson)`, causing the modified state of the address book after the `addnote T0123456A …` command executes to be saved.
+Step 2. The user executes `addnote T0123456A n\Covid` to add a note to the person in ClinicMate with the unique identification number `T0123456A`.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: identification number to IC number

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -260,38 +274,42 @@ The following activity diagram summarizes what happens when a user executes a ne
...
 
 These operations are exposed in the `Model` interface as `Model#updateFilteredPersonList(Predicate predicate)`.
+`predicate` takes in a `IdentityCardNumberMatchesPredicate` to filter the list of patients.
 
 Given below is an example usage scenario and how the find mechanism behaves at each step.
 
 Step 1. The user launches the application. The `AddressBook` will be initialized with the initial address book state.
 
-Step 2. The user executes `find T0123456A …` to find the person in the address book with the unique identification number `T0123456A`. The find command calls `Model#updateFilteredPersonList(Predicate predicate)`, causing the modified state of the address book after the `find T0123456A …` command executes to be displayed.
+Step 2. The user executes `find S0123456A` to find the person in the address book with the unique identification number `S0123456A`. The find command calls `Model#updateFilteredPersonList(Predicate predicate)`,
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: identification number -> IC number

Comment (tengcharmaine) [reply docs/DeveloperGuide.md]: put find into a code block in the 2nd line

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -317,28 +335,36 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 #### Design Considerations & Alternatives Considered
 
-**Aspect: Display of filtered list/contact when command is successful:**
-* Current choice: Displays the correct patient’s contact in the addressbook. 
+**Aspect: Filtering patients**
+* **Alternative 1 (Current Choice):** Requires user to input a full and valid IC number
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: users?

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -317,28 +335,36 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+  * Pros: Precise results, allows user to directly single out the patient details
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: users?

Comment (tengcharmaine) [reply docs/DeveloperGuide.md]: patient's

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -317,28 +335,36 @@ The following activity diagram summarizes what happens when a user executes a ne
...
 
 These operations are exposed in the `Model` interface as `Model#deletePerson(Person target)`.
 
 Given below is an example usage scenario and how the delete mechanism behaves at each step.
 
 Step 1. The user launches the application. The `AddressBook` will be initialized with the initial address book state.
 
-Step 2. The user executes `delete T0123456A` to delete the person in the address book with the unique identification number `T0123456A`. 
-The delete command calls `Model#deletePerson(Person target)`, causing the modified state of the address book after the `delete T0123456A` command executes to be saved.
+Step 2. The user executes `delete S0123456A` to delete the person in the address book with the unique identification number `S0123456A`.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: identification number -> IC number

**Review by tengcharmaine — APPROVED** [review]
LGTM

**Pull request merged**